File: hackingWifi/c/stream/streamServer.c
#include <stdio.h>
#include <pthread.h>
#include "streamSocket.h"
extern void beforeEnd(int ,void *);
void handlerSignalSIGINT(int signal){
	exit(-1);
}
void *handlerClient(void *args){
	Stream client=*((Stream*)args);
	char *string=NULL;
	do{
		string=(char *)receiveObject(client.socket);
		printf("cliente ip:%s puerto:%d dice -> %s\n",inet_ntoa(client.address.sin_addr),client.address.sin_port,string);
	}while(strcmp(string,"-1")!=0);
	printf("cliente %s/%d finalizo su connecion\n",inet_ntoa(client.address.sin_addr),client.address.sin_port);
	return NULL;
}
void waitClientsForEver(Stream connection){
	Stream connectionClient;
	socklen_t soac=sizeof(struct sockaddr_in);//sizeof address client
	while(1){
		printf("esperando cliente ...\n");
		if((connectionClient.socket=accept(connection.socket,(struct sockaddr *)&connectionClient.address,&soac))<0){
			printf("error al aceptar la conexion con el cliente\n");
			exit(-1);
		}
		printf("cliente conectado desde ip:%s puerto:%d\n",inet_ntoa(connectionClient.address.sin_addr),connectionClient.address.sin_port);
		pthread_t thread;
		pthread_create(&thread,NULL,handlerClient,(void *)&connectionClient);
		sleep(1);
	}
}
int main(int ari,char **arc){//./server port maximumConnections
	if(ari<3){
		printf("ejecute como %s puerto maximoConeccionesParaAceptar\n",arc[0]);
		return 0;
	}
	Stream connection=getStreamServerConnection(atoi(arc[1]),atoi(arc[2]));
	waitClientsForEver(connection);
	return 0;
}

File: hackingWifi/java/stream/StreamServer.java
package stream;
import java.net.ServerSocket;
import java.net.Socket;
import java.io.ObjectInputStream;
import java.io.ObjectOutputStream;
public class StreamServer{
	private ServerSocket socket;
	private Socket client;
	private Runnable threadHandlerClient=new Runnable(){
		public void run(){
			receiveCommands();
		}
	};
	public StreamServer(int port){
		try{
			socket=new ServerSocket(port);
			socket.setReuseAddress(true);
			System.out.print("servidor iniciado en puerto:"+port+"\n");
		}catch(Exception e){
			System.out.print("error iniciando servidor en puerto: "+port+"\t"+e.toString()+"\n");
		}
	}
	public boolean sendObject(Socket socket,Object object){
		try{
			ObjectOutputStream oos=new ObjectOutputStream(socket.getOutputStream());
			oos.writeObject(object);
		}catch(Exception e){
			System.out.print("error enviando objecto del socket ip:"+socket.getInetAddress().getHostAddress()+" puerto:"+socket.getPort()+"\t"+e.toString()+"\n");
			return false;
		}
		return true;
	}
	public Object receiveObject(Socket socket){
		Object object=null;
		try{
			ObjectInputStream ois=new ObjectInputStream(socket.getInputStream());
			object=ois.readObject();
		}catch(Exception e){
			System.out.print("error reciviendo objecto del socket ip:"+socket.getInetAddress().getHostAddress()+" puerto:"+socket.getPort()+"\t"+e.toString()+"\n");
		}
		return object;
	}
	public void receiveCommands(){
		Socket client=this.client;
		String string;
		do{
			string=(String)receiveObject(client);
			System.out.print("ip:"+client.getInetAddress().getHostAddress()+" port:"+client.getPort()+" dice -> "+string+"\n");
		}while(!string.equals("-1"));
	}
	public void waitClients(){
		try{
			while(true){
				System.out.print("esperando cliente...\n");
				client=socket.accept();
				System.out.print("cliente conectado desde ip:"+client.getInetAddress().getHostAddress()+" puerto:"+client.getPort()+"\n");
				Thread thread=new Thread(threadHandlerClient);
				thread.start();
				Thread.sleep(500);
			}
		}catch(Exception e){
			System.out.print("error esperando clientes\t"+e.toString()+"\n");
		}
	}
}
